src/notiDaemon/notiDaemon.vala

```diff
@@ -242,6 +243,85 @@ namespace SwayNotificationCenter {
             return id;
         }
 
+        // TODO: Having to reconstruct these params, and the duplication of the above code, is really bad.
+        // Trying to just pass the NotifyParams results in a compilation error, though :-(
+        public void run_action_scripts(
+            uint32 id,
+            string app_name,
+            uint32 replaces_id,
+            string app_icon,
+            string summary,
+            string body,
+            string[] actions,
+            HashTable<string, Variant> hints,
+            int expire_timeout) throws DBusError, IOError {
```

Comment:
To make your life easier, I'd recommend only requiring a `NotifyParams` when calling the method :)

Reply:
Something like this:

```vala
[DBus (visible = false)]
public void run_action_scripts(NotifyParams param) throws DBusError, IOError {
```

The `[DBus (visible = false)]` ensures that the method won't be exported to the DBUS interface so using `NotifyParams` works as expected

Reply:
Oh, I didn't know about that! That should solve most of the egregious issues - I'll go ahead and fix that up.